sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/GitHubPackagesHelper.java

```diff
@@ -0,0 +1,112 @@
+package org.openmrs.maven.plugins.utility;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.settings.Profile;
+import org.apache.maven.settings.Repository;
+import org.apache.maven.settings.RepositoryPolicy;
+import org.apache.maven.settings.Server;
+import org.openmrs.maven.plugins.model.DistroProperties;
+import org.openmrs.maven.plugins.model.GithubPackage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.List;
+
+/**
+ * Helper class for managing GitHub Packages configuration in Maven settings.xml
+ */
+public class GitHubPackagesHelper {
+    
+    private static final Logger log = LoggerFactory.getLogger(GitHubPackagesHelper.class);
+
+    private final SettingsManager settingsManager;
+
+    private static final String GITHUB_MAVEN_URL = "https://maven.pkg.github.com/%s/%s";
```

Comment:
When we talk about supporting "Github Packages" in the SDK, is this about supporting github packages acting as a maven repository?  If so, do we need all of this, or can we find a way for this to work without any of this extra code?

Reply:
To support github packages, one has to update their settings.xml file with certain requirements as defined in this link -> https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry.

What i am attempting here is to do that for the user, updating the settings.xml file as long as they have provided me with the username, token, repository and owner.

Reply:
Honestly, I am not sure I am overly supportive of adding this to the SDK.  I think it is perfectly reasonable for those that use the SDK with github package dependencies to have to set up their settings.xml appropriately and not have the SDK do this.

Reply:
That's a fair point, and I actually agree that while automation through the SDK might improve convenience, it introduces complexity and potential security concerns. Requiring users to configure their `settings.xml` explicitly keeps things more transparent and aligns with Maven best practices.

I will go ahead and close this PR and the ticket as won't fix